processor/src/decorators/mod.rs

```diff
@@ -121,6 +125,24 @@ impl Process {
 
         Ok(())
     }
+
+    /// Sends read request to the memory chiplet to read the specfied number of words from the
+    /// memory starting at the given start address and writes the vector of field elements to
+    /// the advice map with the top 4 elements on the stack as the key.
+    ///
+    /// # Errors
+    /// Returns an error if the key is already present in the advice map.
+    fn inject_mem_values(&mut self, start_addr: u32, num_words: u32) -> Result<(), ExecutionError> {
+        let ctx = self.system.ctx();
+        let mut values = Vec::new();
```

Comment:
nit: since we know the number of elements, we could use `Vec::with_capacity()` here.